elastic/elastic-agent
# Fix regex to locate elastic-agent .zip packages in package tests

## What does this PR do?
This PR fixes the regular expression used to find elastic-agent .zip packages and test them.


## Why is it important?
We want to run tests on freshly generated packages


## Checklist




## Author's Checklist



## How to test this PR locally


To test the PR we can add a blatantly wrong assertion like `assert.True(t, false)` in the `checkZip` function in `package_test.go`
and then run 
```shell
SNAPSHOT=true DEV=true PLATFORMS="windows/amd64" PACKAGES="zip"  mage -v package
```

Without this fix, the packages tests pass anyways because we are not matching any file.
With the fix we should see a beautiful test failure like this:
```
=== RUN   TestZip
=== RUN   TestZip/elastic-agent-8.13.0-SNAPSHOT-windows-x86_64.zip_config_file_permissions
=== RUN   TestZip/elastic-agent-8.13.0-SNAPSHOT-windows-x86_64.zip_manifest_file_permissions
=== RUN   TestZip/elastic-agent-8.13.0-SNAPSHOT-windows-x86_64.zip_modules.d_file_permissions
=== RUN   TestZip/License_file_LICENSE.txt
=== RUN   TestZip/License_file_NOTICE.txt
=== NAME  TestZip
    package_test.go:177:
                Error Trace:    /home/paolo/dev/elastic-agent/dev-tools/packaging/package_test.go:177
                                                        /home/paolo/dev/elastic-agent/dev-tools/packaging/package_test.go:91
                Error:          Should be true
                Test:           TestZip
--- FAIL: TestZip (0.01s)
    --- PASS: TestZip/elastic-agent-8.13.0-SNAPSHOT-windows-x86_64.zip_config_file_permissions (0.00s)
    --- PASS: TestZip/elastic-agent-8.13.0-SNAPSHOT-windows-x86_64.zip_manifest_file_permissions (0.00s)
    --- PASS: TestZip/elastic-agent-8.13.0-SNAPSHOT-windows-x86_64.zip_modules.d_file_permissions (0.00s)
    --- PASS: TestZip/License_file_LICENSE.txt (0.00s)
    --- PASS: TestZip/License_file_NOTICE.txt (0.00s)
=
```


## Related issues


- 

## Use cases



## Screenshots



## Logs



## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...
